bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -218,40 +214,30 @@ public void dispose() {
         super.dispose();
     }
 
-    private boolean checkConnection() {
+    private void connect() {
         @Nullable
-        OpenThermGatewayConnector conn = connector;
+        OpenThermGatewayConfiguration conf = config;
 
-        if (conn != null && conn.isConnected()) {
-            return true;
-        }
+        explicitDisconnect = false;
 
-        return connect();
-    }
-
-    private boolean connect() {
-        @Nullable
-        OpenThermGatewayConfiguration conf = config;
+        if (connecting) {
```

Comment:
Where is this set?

Reply:
It's set to true by the connecting() callback function that is invoked by OpenThermGatewaySocketConnector during run(). When a connection is established, it's then set to false by the connected() callback function.